lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4432,14 +4432,14 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
             GetNegPosInstTest(TestIsSubnormal, FPClass & fcNegSubnormal)));
     }
     if (FPClass & fcZero) {
-      // Create zero integer constant and check for equality with bitcasted to
+      // Create zero integer constant and check for equality with converted to
       // int float value
```

Comment:
Not sure if that works correctly for e.g. an input value `0.123`, as `OpConvertFToU` converts that to `i32 0` so that the compare with 0 succeeds.  Effectively making the emulated fpclass report that `0.123` is zero.  Or did I miss something?

Reply:
You are right, back to the drawing board